src/tests/RegisterPatientPage.test.tsx

```diff
@@ -0,0 +1,85 @@
+import React from 'react'
+import RegisterPatientPage from "../pages/RegisterPatientPage";
+import {render, screen, waitFor} from "@testing-library/react";
+import user from '@testing-library/user-event'
+import {RegistrationData} from "../components/RegisterPatientForm";
+import {setupServer} from "msw/node";
+import {MockedRequest, ResponseComposition, rest} from "msw";
+import {SubmitHandler} from "react-hook-form";
+
+const mockedData = {
+    address: {
+        city: "Warsaw",
+        houseNumber: "11A",
+        localNumber: "3",
+        street: "Ogrodowa",
+        zipCode: "12-123",
+    },
+    email: "abc@wp.pl",
+    firstName: "adam",
+    lastName: "abacki",
+    password: "abc",
+    pesel: "98071962617",
+}
+
+const server = setupServer(
+    rest.post('*/patient/registration', async (request: MockedRequest, response: ResponseComposition, ctx) => {
+        // returning the sent object when successful
+        return response(
+            ctx.delay(100),
+            ctx.json(request.body));
+    })
+);
+
+beforeAll(() => server.listen());
+afterAll(() => server.close());
+afterEach(() => server.resetHandlers());
+
+jest.mock("../components/RegisterPatientForm", () => {
```

Comment:
Jak to wsm działa?

Reply:
Mockuje komponent z ../components/RegisterPatientForm czyli <RegisterPatientForm/> jako inny komponent. I teraz jak będzie on wywoływany przy okazji render to zamiast niego będzie tworzony ta zdefiniowana makietka. Źródło: https://reactjs.org/docs/testing-recipes.html#mocking-modules